### ISSUE TITLE
(tLN) Supervision function selector scoping

### ISSUE THREAD

**Issue by danyill** [OpenEnergyTools/scl-lib#6]
The selectors we use in our supervision functions are too broad:

```js
/** @returns Element to remove the subscription supervision */
function removableSupervisionElement(
  ctrlBlock: Element,
  subscriberIed: Element
): Element | null {
  const supervisionType = ctrlBlock.tagName === "GSEControl" ? "LGOS" : "LSVS";

  const valElement = Array.from(
    subscriberIed.querySelectorAll(
      `LN[lnClass="${supervisionType}"] > DOI > DAI > Val`
    )
  ).find((val) => val.textContent === controlBlockObjRef(ctrlBlock));
  if (!valElement) return null;

  const ln = valElement.closest("LN")!;
  const doi = valElement.closest("DOI")!;

  // do not remove logical nodes `LGOS`, `LSVS` unless privately tagged
  const canRemoveLn = ln.querySelector(
    ':scope > Private[type="OpenSCD.create"]'
  );

  return canRemoveLn ? ln : doi;
}
```

This is too broad: `LN[lnClass="${supervisionType}"] > DOI > DAI > Val`

The DAI must be scoped to `setSrcRef` and the DOI must be scoped to `GoCBRef` or `SvCBRef`

I have generally improved these as I've noticed this in a few places: https://github.com/danyill/oscd-subscriber-later-binding/blob/main/foundation/subscription/subscription.ts

I can do a PR to fix this if you agree.

**Comment by JakobVogelsang** [comment]
Very true. Thanks for spotting. If you want to go ahead with a PR.

**Comment by danyill** [comment]
I'll have a go at doing this and adding some tests to get started here.

**Comment by danyill** [comment]
Resolved via #84, so this can be closed
